hw4.js

```diff
@@ -0,0 +1,231 @@
+let offset = +prompt('the end of the letter number');
+
+function trimString(string, from, to) {
+    return string.slice(from, to)
+};
+
+alert(trimString(text, onset, offset));
+
+// #### Task 4 💻
+
+// Написать функцию **getSumNumbers**, которая будет принимать число и вычислять сумму цифр из которых состоит число.
+// > Для 2021 это будет 5.
+
+// FIRST SOLUTION
+
+let number = prompt('insert number');
+
+function getSumNumbers(num) {
+    return num.split('').reduce((a, b) => +a + +b, 0)
+};
+
+alert(getSumNumbers(number));
+
+// SECOND SOLUTION
+
+let numberTwo = +prompt('insert number');
+let sum = 0;
+
+function getSumNumbersSecond(num) {
+    let array = num.toString().split('')
+    for(let i = 0; i < array.length; i++) {
+        sum += +array[i]
+    }
+    return sum
+};
+alert(getSumNumbersSecond(numberTwo));
+
+// #### Task 5 💻
+
+// Написать функцию **`getSum`** которая принимает два целых числа a и b, 
+// которые могут быть положительными или отрицательными, найти сумму всех чисел между ними, 
+// включая их, и вернуть ее. Если два числа равны, верните a или b.
+
+// ```
+//     getSum(1, 0) == 1   // 1 + 0 = 1
+//     getSum(1, 2) == 3   // 1 + 2 = 3
+//     getSum(0, 1) == 1   // 0 + 1 = 1
+//     getSum(1, 1) == 1   // 1 Since both are same
+//     getSum(-1, 0) == -1 // -1 + 0 = -1
+//     getSum(-1, 2) == 2  // -1 + 0 + 1 + 2 = 2
+// ```
+
+let numbOne = +prompt("Введите первое число");
+let numbTwo = +prompt("Введите второе число");
+
+function getSum(num1, num2) {
+    let result = num1
+    if (num1 > num2) {
+        for (let i = num2; i < num1; i++) {
+        result += i;
+        }
+    return result
+    } else if (num1 < num2) {
+        for (let i = num1; i < num2; i++) {
+        result += i+1;
+        }
+    } else result = `Вы дважды ввели число ${num1}`
+    return result
+};
+
+alert(getSum(numbOne, numbTwo));
+
+// #### Task 6 💻
+
+// Напишите функцию **fooboo** которая принимает в качестве аргумента три параметра:
+
+// + булевое значение
+// + функцию **foo** которая выводит в консоль свое имя
+// + функцию **boo** которая выводит в консоль свое имя
+
+// > Если переданное булевое значение **true** запускаем функцию **foo** иначе **boo**
+
+
+
+function fooboo(par, par2, par3) {
+    if(!par) {
+        par3()
+    } else par2()
+};
+
+function foo() {
+    console.log('foo');
+};
+
+function boo() {
+    console.log('boo');
+};
+
+console.log(fooboo(12, foo, boo));
+
+
+// ### ADVANCED level
+
+// #### Task 1 👨‍🏫 
+
+// + Реализуйте функцию, который принимает 3 целочисленных значения a, b, c. Функция должна возвращать **true**,
+//  если треугольник можно построить со сторонами заданной длины, и **false** в любом другом случае.
+
+let angle1 = +prompt('insert first angle lenght');
+let angle2 = +prompt('insert second angle lenght');
+let angle3 = +prompt('insert third angle lenght');
+
+function triangle(a, b, c) {
+    if ((a < (b + c)) && (b < (c + a)) && (c < (a + b))) {
```

Comment:
you can return (a < (b + c)) && (b < (c + a)) && (c < (a + b)); without condiotions here.